planckton/sim.py

```diff
@@ -129,138 +129,134 @@ def __init__(
+        )
+        sim.operations.writers.append(gsd_writer)
+        sim.operations.writers.append(table_file)
 
-            hoomd.dump.gsd(
-                filename="trajectory.gsd",
-                period=self.gsd_write,
-                group=all_particles,
-                overwrite=False,
-                phase=0,
-                dynamic=["momentum"],
+        if self.target_length is not None:
+            # Run the shrink step
+            final_length = self.target_length.to("Angstrom").value
+            final_box = (self.shrink_steps, final_length)
+            box_resize_trigger = hoomd.trigger.Periodic(self.shrink_period)
+            ramp = hoomd.variant.Ramp(
+                A=0, B=1, t_start=0, t_ramp=int(self.shrink_steps)
             )
-            gsd_restart = hoomd.dump.gsd(
-                "restart.gsd",
-                period=self.gsd_write,
-                group=all_particles,
-                truncate=True,
-                phase=0,
-                dynamic=["momentum"],
+            initial_box = sim.state.box
+            final_box = hoomd.Box(
+                Lx=final_length,
+                Ly=final_length,
+                Lz=final_length,
             )
-            log_quantities = [
-                "temperature",
-                "pressure",
-                "volume",
-                "potential_energy",
-                "kinetic_energy",
-                "pair_lj_energy",
-                "bond_harmonic_energy",
-                "angle_harmonic_energy",
-            ]
-            hoomd.analyze.log(
-                "trajectory.log",
-                quantities=log_quantities,
-                period=self.log_write,
-                header_prefix="#",
-                overwrite=False,
-                phase=0,
+            box_resize = hoomd.update.BoxResize(
+                box1=initial_box,
+                box2=final_box,
+                variant=ramp,
+                trigger=box_resize_trigger,
             )
-
-            if self.target_length is not None:
-                # Run the shrink step
-                final_length = self.target_length.to("Angstrom").value
-                final_box = (self.shrink_steps, final_length)
-                size_variant = hoomd.variant.linear_interp(
-                    [(0, snap.box.Lx), final_box], zero=0
-                )
-                box_resize = hoomd.update.box_resize(L=size_variant)
-                integrator.randomize_velocities(seed=42)
-                hoomd.run_upto(self.shrink_steps)
-                box_resize.disable()
-                self.n_steps = [i + self.shrink_steps for i in self.n_steps]
+            sim.operations.updaters.append(box_resize)
+            sim.state.thermalize_particle_momenta(
+                filter=all_particles, kT=self.shrink_kT
+            )
+            sim.run(self.shrink_steps, write_at_start=True)
+            self.n_steps = [i + self.shrink_steps for i in self.n_steps]
```

Comment:
Not sure if I understood the purpose of `n_steps` correctly but if `self.n_steps` is a list of number of steps for each simulation block, it means the values in that list indicate the length of each simulation block, not the final timestep at each block. In that case, you don't need to add the `shrink_steps` here.